src/bin/ion/commands/beta/inspect.rs

```diff
@@ -120,613 +141,1237 @@ complete value will be displayed.",
+            // offset fell at the beginning of a top level value, the line below will set the flag
+            // for the first time.
+            self.skip_complete = true;
+            is_first_item = false;
         }
-        if let Some(offset) = self.reader.annotations_offset() {
-            return offset;
+        self.output.write_all(END_OF_TABLE.as_bytes())?;
+        Ok(())
+    }
+
+    /// If `maybe_item` is:
+    ///    * `Some(entity)`, checks to see if the entity's final byte offset is beyond the configured
+    ///                      number of bytes to skip.
+    ///    * `None`, then there is no stream-level entity backing the item (that is: it was the result
+    ///              of a macro expansion). Checks to see if the inspector has already completed its
+    ///              skipping phase on an earlier item.
+    fn should_skip<T: HasRange>(&mut self, maybe_item: &Option<T>) -> bool {
+        match maybe_item {
+            // If this item came from an input literal, see if the input literal ends after
+            // the requested number of bytes to skip. If not, we'll move to the next one.
+            Some(item) => item.range().end <= self.bytes_to_skip,
+            // If this item came from a macro, there's no corresponding input literal. If we
+            // haven't finished skipping input literals, we'll skip this ephemeral value.
+            None => !self.skip_complete,
         }
-        self.reader.header_offset()
     }
 
-    // Returns the byte offset range containing the current value and its annotations/field ID if
-    // applicable.
-    fn complete_value_range(&self) -> Range<usize> {
-        let start = self.first_value_byte_offset();
-        let end = self.reader.value_range().end;
-        start..end
+    /// If `maybe_item` is:
+    ///    * `Some(entity)`, checks to see if the entity's final byte offset is beyond the configured
+    ///                      number of bytes to inspect.
+    ///    * `None`, then there is no stream-level entity backing the item. These will always be
+    ///              inspected; if the e-expression that produced the value was not beyond the limit,
+    ///              none of the ephemeral values it produces are either.
+    fn is_past_limit<T: HasRange>(&mut self, maybe_item: &Option<T>) -> bool {
+        let limit = self.bytes_to_skip.saturating_add(self.limit_bytes);
+        maybe_item
+            .as_ref()
+            .map(|item| item.range().start >= limit)
+            .unwrap_or(false)
+    }
+
+    /// Convenience method to set the output stream to the specified color/style for the duration of `write_fn`
+    /// and then reset it upon completion.
+    fn with_style(
+        &mut self,
+        style: ColorSpec,
+        write_fn: impl FnOnce(&mut OutputRef) -> Result<()>,
+    ) -> Result<()> {
+        self.output.set_color(&style)?;
+        write_fn(&mut self.output)?;
+        self.output.reset()?;
+        Ok(())
     }
```

Comment:
I'm a little surprised this doesn't already exist in `termcolor`, since this is usually the idiom I reach for for colorized text. 
Indeed, though, it's not there: https://docs.rs/termcolor/latest/termcolor/trait.WriteColor.html